app/src/main/java/com/epfl/drawyourpath/utils/Utils.kt

```diff
@@ -392,4 +394,118 @@ object Utils {
+     * @param start The starting point of the segment
+     * @param end The end point of the segment
+     * @return The distance between each point and the segment.
+     */
+    private fun distancesToSegment(points: List<LatLng>, start: LatLng, end: LatLng): List<Float> {
+        // Translate to put the start at the origin
+        var newEnd = LatLng(end.latitude - start.latitude, end.longitude - start.longitude)
+        // Consider the angle in the plane perpendicular to the x axis in spherical coordinates
+        // So the angle is atan2(z, y) where z = sin(latitude), y = cos(latitude)sin(longitude)
+        val anglePlane = atan2(
+            sin(newEnd.latitude * PI / 180),
+            cos(newEnd.latitude * PI / 180) * sin(newEnd.longitude * PI / 180),
+        )
+        // Rotate the end point to be aligned with the equator
+        newEnd = rotateXAxis(newEnd.latitude, newEnd.longitude, -anglePlane)
+
+        var distances = mutableListOf<Float>()
+        for (point in points) {
+            // Translate to put the start at the origin and rotate to put the segment on the equator
+            val newPoint = rotateXAxis(
+                point.latitude - start.latitude,
+                point.longitude - start.longitude,
+                -anglePlane,
+            )
+            // Compute the distance to the segment case by case
+            var results = FloatArray(3)
+            if (newPoint.longitude < 0) {
+                Location.distanceBetween(start.latitude, start.longitude, point.latitude, point.longitude, results)
+            } else if (newPoint.longitude > newEnd.longitude) {
+                Location.distanceBetween(end.latitude, end.longitude, point.latitude, point.longitude, results)
+            } else {
+                Location.distanceBetween(0.0, 0.0, newPoint.latitude, 0.0, results)
+            }
+            distances.add(results[0])
+        }
+        return distances.toList()
+    }
+
+    /**
+     * Compute the rotation around the X axis given the latitude and longitude.
+     * @param latitude The latitude of the input point
+     * @param longitude The longitude of the input point
+     * @param angle The angle of rotation (in radians)
+     * @return The point after applying the rotation.
+     */
+    private fun rotateXAxis(latitude: Double, longitude: Double, angle: Double): LatLng {
+        val latRadian = latitude * PI / 180
+        val lonRadian = longitude * PI / 180
+        // Transform to 3D coordinates
+        val x = cos(latRadian) * cos(lonRadian)
+        val y = cos(latRadian) * sin(lonRadian)
+        val z = sin(latRadian)
+        // Apply rotation along the X axis (equivalent to a 2D rotation on y and z)
+        val newX = x
+        val newY = cos(angle) * y - sin(angle) * z
+        val newZ = sin(angle) * y + cos(angle) * z
+        // Transform back to latitude and longitude
+        val newLat = asin(newZ)
+        val newLon = atan2(newY, newX)
```

Comment:
The variable `newX` is basically useless, you can just keep `x`
```suggestion
        val newY = cos(angle) * y - sin(angle) * z
        val newZ = sin(angle) * y + cos(angle) * z
        // Transform back to latitude and longitude
        val newLat = asin(newZ)
        val newLon = atan2(newY, x)
```

Reply:
Yes as the rotation is only around the X axis the x coordinate does not change.
I'll remove the useless variable and put a comment instead